lib/internal/bootstrap/pre_execution.js

```diff
@@ -178,6 +179,13 @@ function setupDebugEnv() {
   if (getOptionValue('--expose-internals')) {
     require('internal/bootstrap/loaders').NativeModule.exposeInternals();
   }
+
+  const { builtInObjects } = require('internal/util/inspect');
+  // The content of this list depends on e.g. v8 flags, as well as
+  // what gets added by Node.js to the global object.
+  for (const key of ObjectGetOwnPropertyNames(globalThis)) {
```

Comment:
This now contains these added by Node.js

```
  'global',
  'process',
  'Buffer',
  'URL',
  'URLSearchParams',
  'TextEncoder',
  'TextDecoder',
  'clearInterval',
  'clearTimeout',
  'setInterval',
  'setTimeout',
  'queueMicrotask',
  'clearImmediate',
  'setImmediate'
```

We still need this loop if we want to pick up what get's added by V8 through harmony flags (like WeakRefs). I wonder which of these do we want to skip? @BridgeAR

Reply:
We mostly do not need those. The only ones that we should keep are global classes that have a prototype. The current RegExp is not matching exactly what we actually need. I just checked and this seems to ideally model what is needed for inspect:

```js
function isConstructor(name) {
  try {
    Reflect.construct(String, [], globalThis[name]);
  } catch {
    return false;
  }
  return globalThis[name].prototype !== undefined && name[0].toUpperCase() === name[0];
}
```